src/fclone.js

```diff
@@ -1,5 +1,13 @@
 'use strict';
 
+// see if it looks and smells like an iterable object, and do accept length === 0
+function isArrayLike(item) {
+  if (Array.isArray(item)) return true;
+
+  var len = item && item.length;
+  return typeof len === 'number' && (len === 0 || (len - 1) in item);
```

Comment:
If you have an object `{length:5}`, without a property of `4` it isn't considered array-like.  It just checks that if it isn't an array, then the key exists...